pkg/build/controller/strategy/util.go

```diff
@@ -68,47 +69,63 @@ func setupDockerSocket(podSpec *v1.Pod) {
 		MountPath: dockerSocketPath,
 	}
 
-	podSpec.Spec.Volumes = append(podSpec.Spec.Volumes,
+	pod.Spec.Volumes = append(pod.Spec.Volumes,
 		dockerSocketVolume)
-	podSpec.Spec.Containers[0].VolumeMounts =
-		append(podSpec.Spec.Containers[0].VolumeMounts,
+	pod.Spec.Containers[0].VolumeMounts =
+		append(pod.Spec.Containers[0].VolumeMounts,
 			dockerSocketVolumeMount)
+	for i, initContainer := range pod.Spec.InitContainers {
+		if initContainer.Name == "extract-image-content" {
+			pod.Spec.InitContainers[i].VolumeMounts = append(pod.Spec.InitContainers[i].VolumeMounts, dockerSocketVolumeMount)
+			break
+		}
+	}
 }
 
 // mountSecretVolume is a helper method responsible for actual mounting secret
 // volumes into a pod.
-func mountSecretVolume(pod *v1.Pod, secretName, mountPath, volumeSuffix string) {
+func mountSecretVolume(pod *v1.Pod, container *v1.Container, secretName, mountPath, volumeSuffix string) {
 	volumeName := namer.GetName(secretName, volumeSuffix, kvalidation.DNS1123SubdomainMaxLength)
-	volume := v1.Volume{
-		Name: volumeName,
-		VolumeSource: v1.VolumeSource{
-			Secret: &v1.SecretVolumeSource{
-				SecretName: secretName,
+	volumeExists := false
+	for _, v := range pod.Spec.Volumes {
+		if v.Name == volumeName {
+			volumeExists = true
+			break
+		}
+	}
+	if !volumeExists {
```

Comment:
Previously there was no risk of us adding the same volume to a pod multiple times because there was only one container.  Now there is because we add the volume+volumemount once for each container.